runtime/oti/j9nonbuilder.h

```diff
@@ -6078,9 +6078,9 @@ typedef struct J9JavaVM {
 	UDATA contiguousIndexableHeaderSize;
 	UDATA discontiguousIndexableHeaderSize;
 #if defined(J9VM_ENV_DATA64)
-	UDATA isIndexableDataAddrPresent;
-	BOOLEAN isVirtualLargeObjectHeapEnabled;
-	BOOLEAN isIndexableDualHeaderShapeEnabled;
+	U_32 isIndexableDataAddrPresent;
+	U_32 isVirtualLargeObjectHeapEnabled;
+	U_32 isIndexableDualHeaderShapeEnabled;
```

Comment:
Type changes like these cause compatibility problems for DDR accessing older system dumps where the fields had different types. See my earlier comment: https://github.com/eclipse-openj9/openj9/pull/20804#pullrequestreview-2528093377.

Reply:
OK, I see that javaVM.isIndexableDataAddrPresent() was indeed explicitly used in DDR. I don't see explicit usages of the other 2 fields, though.

Soon, we will be removing all these 3 fields and replacing them with one composite field (a bit-wise union), at which point we will be forced to properly implement isIndexableDataAddrPresent() to extract the value from the composite field or try to return the field itself, if it exists, for older cores.

That should resolve potential compatibility problems between DDR and cores for released JVMs (although some non-released JVMs could still experience incompatibilities).

Reply:
This only serves to complicate matters as now there are (at least) two kinds of "older cores". It would have been better to leave the fields as they were and switch to the bit-mask in one step.

Reply:
We haven't shipped this change yet, until the 0.51 release.